src/services/etherspot.js

```diff
@@ -61,6 +69,73 @@ class EtherspotService {
       });
   }
 
+  async getBalances(accountAddress: string, assets: Asset[]): Promise<Balance[]> {
+    const assetAddresses = assets
+      // 0x0...0 is default ETH address in our assets, but it's not a token
+      .filter(({ address }) => !isCaseInsensitiveMatch(address, EthersConstants.AddressZero))
+      .map(({ address }) => address);
+
+    let balancesRequestPayload = {
+      account: accountAddress,
+    };
+
+    if (assetAddresses.length) {
+      balancesRequestPayload = {
+        ...balancesRequestPayload,
+        tokens: assetAddresses,
+      };
+    }
+
+    // gets balances by provided token (asset) address and ETH balance regardless
+    const accountBalances = await this.sdk
+      .getAccountBalances(balancesRequestPayload)
+      .catch((error) => {
+        reportErrorLog('EtherspotService getBalances -> getAccountBalances failed', { error, accountAddress });
+        return null;
+      });
+
+    if (!accountBalances) {
+      return []; // logged above, no balances
+    }
+
+    // map to our Balance type
+    return accountBalances.items.reduce((balances, { balance, token }) => {
```

Comment:
readability: Consider replacing `reduce` with `const result = []` + `accountsBalances.items.forEach` for somewhat less complex code.

Reply:
Hm, I think this one is by preference, I prefer `reduce` over `forEach` and appending to array outside loop 🤔